projects/gnoland/gno.land/p/eve000/event/agenda/agenda.gno

```diff
@@ -58,8 +58,14 @@ func (a *Agenda) ToMarkdown(_ ...component.Content) string {
 
 	markdown += a.Description + "\n\n"
 
-	if a.Banner != nil && a.Banner.Published {
-		markdown += a.Banner.Markdown + "\n\n"
+	if len(body) > 0 {
+		for _, content := range body {
+			if content.Published {
+				markdown += content.Render() + "\n\n"
+			}
+		}
+	} else if a.Banner != nil && a.Banner.Published {
+		markdown += a.Banner.Render() + "\n\n"
```

Comment:
could optionally remove Published check here since we only return empty string if unpublished